modules/apps/27-interchain-accounts/host/keeper/relay.go

```diff
@@ -66,27 +66,26 @@ func (k Keeper) executeTx(ctx context.Context, sourcePort, destPort, destChannel
 		MsgResponses: make([]*codectypes.Any, len(msgs)),
 	}
 
-	// CacheContext returns a new context with the multi-store branched into a cached storage object
-	// writeCache is called only if all msgs succeed, performing state transitions atomically
-	sdkCtx := sdk.UnwrapSDKContext(ctx) // TODO: https://github.com/cosmos/ibc-go/issues/5917
-	cacheCtx, writeCache := sdkCtx.CacheContext()
 	for i, msg := range msgs {
 		if m, ok := msg.(sdk.HasValidateBasic); ok {
 			if err := m.ValidateBasic(); err != nil {
 				return nil, err
 			}
 		}
 
-		protoAny, err := k.executeMsg(cacheCtx, msg)
-		if err != nil {
+		if err := k.BranchService.Execute(ctx, func(ctx context.Context) error {
```

Comment:
Heh, I guess the name is marginally better than CacheContext, and this is more typical of how a database transaction interface would look like, but it also would have been nice if there was a function similar to CacheContext to make upgrades smoother. Oh well...

Reply:
yea, thinking this would look clunkier in the `switch` we do in core handlers where we only `write` on one case.